extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java

```diff
@@ -0,0 +1,317 @@
+package io.quarkus.scheduler.runtime;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.annotation.PreDestroy;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Typed;
+import javax.inject.Singleton;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.jboss.logging.Logger;
+import org.jboss.threads.JBossScheduledThreadPoolExecutor;
+
+import com.cronutils.model.Cron;
+import com.cronutils.model.definition.CronDefinition;
+import com.cronutils.model.definition.CronDefinitionBuilder;
+import com.cronutils.model.time.ExecutionTime;
+import com.cronutils.parser.CronParser;
+
+import io.quarkus.runtime.StartupEvent;
+import io.quarkus.scheduler.Scheduled;
+import io.quarkus.scheduler.ScheduledExecution;
+import io.quarkus.scheduler.Scheduler;
+import io.quarkus.scheduler.Trigger;
+
+@Typed(Scheduler.class)
+@Singleton
+public class SimpleScheduler implements Scheduler {
+
+    private static final Logger LOGGER = Logger.getLogger(SimpleScheduler.class);
+
+    private static final long CHECK_PERIOD = 1000L;
+
+    private final ScheduledExecutorService scheduledExecutor;
```

Comment:
One thing I was curious about and I think I mentioned before is, whether it's worth exposing the `ScheduledExecutorService` directly to applications which want to programmatically schedule things.  Maybe as a future enhancement?

Reply:
Yes, that's why I removed the `Scheduler#startTimer()` method which was basically mimicking `ScheduledExecutorService.schedule()`. In fact, we don't expose any `ExecutorService` ATM and some users coming from the EE world could expect some support of `ManagedExecutorService` and friends.